extensions/amp-social-share/1.0/amp-social-share.css

```diff
@@ -20,6 +20,16 @@
   * See https://github.com/ampproject/amphtml/issues/4277 for more details.
   */
 
+amp-social-share:focus-within {
```

Comment:
Hmm. I'm reading [this article](https://developers.google.com/web/fundamentals/web-components/shadowdom) and according to it, simply just doing `amp-social-share:focus` should work just fine. Did it not work for you?

Reply:
Is it because the `amp-social-share` element is no longer itself focusable? I believe this would apply style if the button inside it generated by `SocialShare` is focused.

Reply:
After reviewing the article, we can use `focus` if we use `delegateFocus: true` when creating the shadowRoot.  I tried this in the latest revision and it was working.  If we do not use `delegateFocus: true`, then the host will not receive focus so we will have to use `focus-within`.

I personally like the `focus` api better than `focus-within`.  If we are okay with the change to `delegateFocus:true` then let's move forward this way.  Here's the change: https://github.com/ampproject/amphtml/pull/31994/files#diff-7ff1bd0078702f4c3fb87e0b61261fbaeaaa4726a121ebd9f3324e3a0e631be1R518

The latest revision uses `delegateFocus:true` and `focus`.

~~~

And that's correct the `amp-social-share` element is no longer focusable in `1.0` because we actually focus on an element that's within the shadowRoot!

Reply:
That's `delegatesFocus`, right? TIL. From reading the docs - this is exactly what we need for `amp-social-share`. And `amp-social-share:focus {}` is exactly what we want, I believe. But this wouldn't work for just any AMP element, so we need to make it configurable via `PreactBaseElement` props. I'd call it exactly that `PreactBaseElement['delegatesFocus']`, which would be `false` by default and overriden to `true` for this element.

Reply:
That's a great recommendation, I've updated in the latest revision.